From da1ef0ee01db8343faf9188ef11127b264f23eae Mon Sep 17 00:00:00 2001
From: Roopni Devanathan <quic_rdevanat@quicinc.com>
Date: Tue, 13 Aug 2024 15:32:07 +0530
Subject: [PATCH] wifi: ath12k: KW: Initialize HAL capabilities arguments

Currently, the struct ath12k_wmi_hal_reg_capabilities_ext_arg
is uninitialized before its use. Initialize it to refrain
from Klocwork error.

Patch-dependency: [None]
Patch-work: [None]

Signed-off-by: Roopni Devanathan <quic_rdevanat@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/wmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index edf3022..75820db 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -6804,7 +6804,7 @@ static int ath12k_wmi_ext_hal_reg_caps(struct ath12k_base *soc,
 {
 	struct ath12k_wmi_pdev *wmi_handle = &soc->wmi_ab.wmi[0];
 	struct ath12k_wmi_svc_rdy_ext_parse *svc_rdy_ext = data;
-	struct ath12k_wmi_hal_reg_capabilities_ext_arg reg_cap;
+	struct ath12k_wmi_hal_reg_capabilities_ext_arg reg_cap = {0};
 	int ret;
 	u32 i;
 
-- 
2.34.1

